rclpy/package.xml

```diff
@@ -32,6 +32,7 @@
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
   <test_depend>python3-pytest</test_depend>
+  <test_depend>python3-parameterized</test_depend>
```

Comment:
`python3-parameterized` is not recognized by rosdep, as it hasn't been added to [rosdistro](https://github.com/ros/rosdistro/blob/175ad318707c160ff104c9329c6e3b5bf38a85b3/rosdep/python.yaml#L2726-L2761). You can send a PR to `rosdistro` ([example PR](https://github.com/ros/rosdistro/pull/21800/files)), or use directly pytest parameterized builtins.

Reply:
> `python3-parameterized` is not recognized by rosdep, as it hasn't been added to [rosdistro](https://github.com/ros/rosdistro/blob/175ad318707c160ff104c9329c6e3b5bf38a85b3/rosdep/python.yaml#L2726-L2761). You can send a PR to `rosdistro` ([example PR](https://github.com/ros/rosdistro/pull/21800/files)), or use directly pytest parameterized builtins.

I have created PR for `rosdistro`.  
https://github.com/ros/rosdistro/pull/23197.

Reply:
And, I forgot, `ros2/ci` repo has to be updated too, like this https://github.com/ros2/ci/pull/346 (also, see https://github.com/ros2/ci/pull/354/files).
I don't have powers neither in `rosdistro` nor `ros2/ci`, so you will have to wait for someone else.

Reply:
If this extra dependency can be avoided by using `pytest` please go ahead and do so in order to not introduce new dependencies.

Reply:
@ivanpauno 

> And, I forgot, ros2/ci repo has to be updated too, like this ros2/ci#346 (also, see https://github.com/ros2/ci/pull/354/files).

Thanks for the reminder.

@dirk-thomas 

> If this extra dependency can be avoided by using pytest please go ahead and do so in order to not introduce new dependencies.

Both are OK, but I think 'python3-parameterized' is more friendly for test developer.  
'python3-parameterized' can be used easily with unittest, so a concrete test case can use 'assert*' functions of unittest.TestCase which provides more information, and it supports the callback of member function in class.  
What do you think ?

Reply:
> And, I forgot, ros2/ci repo has to be updated too, like this ros2/ci#346 (also, see https://github.com/ros2/ci/pull/354/files).

Thanks for the reminder.   
But after consideration (different platform), I will use pytest.

Reply:
Ok, I considered adding `python3-parameterized`, because the resulting code looks nicer.
But continuing using vanilla pytest parameterization support is fine.